src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpUsingVisitor.cs

```diff
@@ -93,11 +97,15 @@ protected override void Visit(TagHelperChunk chunk)
             {
                 _foundTagHelpers = true;
 
-                if (!ImportedUsings.Contains(TagHelpersRuntimeNamespace))
+                foreach (var tagHelperRuntimeNamespace in TagHelpersRuntimeNamespaces)
                 {
-                    // If we find TagHelpers then we need to add the TagHelper runtime namespace to our list of usings.
-                    Writer.WriteUsing(TagHelpersRuntimeNamespace);
-                    ImportedUsings.Add(TagHelpersRuntimeNamespace);
+                    if (!ImportedUsings.Contains(tagHelperRuntimeNamespace))
```

Comment:
Drive-by suggestion but could `ImportedUsings` be a `HashSet<string>`? This would then simplify to
``` c#
if (ImportedUsings.Add(tagHelperRuntimeNamespace))
{
    Writer.WriteUsing(tagHelperRuntimeNamespace);
}
```